src/services/functions.py

```diff
@@ -47,3 +48,87 @@ def login_user(username: str, password: str) -> None:
             raise InvalidUsername
         elif not check_password_hash(user['hashed_password'], password):
             raise InvalidCredentials
+
+def list_courses() -> list:
+    """Returns a list of all courses users can choose.
+
+    Returns:
+        list: A list of all courses users can choose
+    """
+    courses = []
+    with get_db_connection(COURSES_DB) as con:
+        rows = con.execute('SELECT course_code FROM courses').fetchall()
+        for row in rows:
+            courses.append(row['course_code'])
+    return courses
+
+def add_course_to_user(username: str, course_code: str) -> None:
+    """Adds a new course to user's course list.
+    
+    If course is already in user's course list, a CourseAlreadySelected exception is raised.
+
+    Args:
+        username (str): user to whom course should be added
+        course_code (str): code of course to be added to user
+    """
+    with get_db_connection(USER_COURSES_DB) as con:
+        user_courses_json = (
+            con
+            .execute('SELECT course_list_data FROM user_courses WHERE username = ?', (username,))
+            .fetchone()
+        )
+        if not user_courses_json:
+            user_course_list = [course_code]
+            new_user_courses_json = json.dumps(user_course_list)
+            con.execute('INSERT INTO user_courses (username, course_list_data) VALUES (?, ?)',
+                        (username, new_user_courses_json))
```

Comment:
the db calls should be inside db classes that understand the structure. It is called the Data Transfer Object. That way, the DB structure can be kept independent of the application data needed. Sometime, there are computed fields in the Business Object, or Domain Object. But the Data Transfer Object is always only the data that needs to be persisted.